### PR TITLE
Bw bug/subnav bug

### PR DESCRIPTION
**Pull request by barbarawuk**
Show correct subnav e.g. new user should show 'Users' in subnav

### CODE REVIEW

File: app/presenters/nav_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          { label: I18n.t('nav.tabs.admin_group_types'),
+            path: %i[admin group_types],
+            controllers: ['admin/group_types', 'admin/user_groups'],
+            actions: %w[index new edit] }
+        ],
+        enabled: @current_user&.admin?
+      }
+    ]
+  end
+
```
Comment (alexrudall) [inline app/presenters/nav_presenter.rb:52]: Could add some more helpers here:

```suggestion
  def nav_active?(nav)
    path_active?(nav) || any_subnavs_active?(nav)
  end

  def path_active?(nav)
    nav[:path] == [@params['action'].to_sym, @params['controller'].to_sym]
  end

  def any_subnavs_active?(nav)
    nav[:subnavs]&.any? { |subnav| subnav_active?(subnav) }
  end
```

File: app/views/shared/_nav.erb [hunk 2]
```diff
@@ -18,7 +18,7 @@
                   link_to(
                     nav[:label],
                     nav[:path],
-                    class: "nav-link #{' active' if current_page?(nav[:path]) || nav[:subnavs]&.any? { |s| current_page?(s[:path]) } }"
+                    class: "nav-link #{' active' if current_page?(nav[:path]) || nav[:subnavs]&.any? { |subnav| @nav_presenter.subnav_active?(subnav) } }"
```
Comment (alexrudall) [inline app/views/shared/_nav.erb:21]: And move this code to the presenter:

```suggestion
    class: "nav-link #{' active' if @nav_presenter.nav_active?(nav) }"
```

**Review by alexrudall — CHANGES_REQUESTED** [review]
This is really great! I've suggested a few more helpers to move more logic out of the view into the presenter.

File: app/presenters/nav_presenter.rb (hunk 1 above)
Comment (barbarawuk) [reply app/presenters/nav_presenter.rb:52]: I've got something very similar in the next PR so it's coming :-)

**Pull request merged**